quilter.py

```diff
@@ -0,0 +1,360 @@
+'''
+quilter.py
+
+Download ALL THE RASTERS!!!
+
+'''
+
+import csv
+import os
+import tempfile
+import zipfile
+import shutil
+import sys
+import requests
+import traceback
+import inspect
+import subprocess
+import argparse
+from pathlib import Path
+
+# User Input:
+# csv file
+# destination folder
+# -m merge flag and name
+# -p project flag and accompanying EPSG:xxxxx designator
+# -d don't compress tif (ignore for aerial imagery)
+
+parser = argparse.ArgumentParser(description='Download helper for\
+                                              raster.utah.gov')
+
+parser.add_argument('csv', help='CSV file of download links.')
+parser.add_argument('destination', 
+                    help='Directory for extracted files (must already exist).')
+
+parser.add_argument('-m', '--merge', dest="name", 
+                    help='Merge downloaded files to specified base name. .tif\
+                    or .shp extensions will be added as appropriate. Requires\
+                    GDAL.')
+parser.add_argument('-p', '--project', dest="crs", help='Reproject merged file\
+                    to specified CRS. Specify CRS like EPSG:x or ESRI:x.\
+                    Requires -m.')
+parser.add_argument('-d', '--dont_compress', help='Disable LZW compression of\
+                    merged raster (all aerials will remain compressed).',
+                    action='store_true')
+
+args = parser.parse_args()
+
+# Set defaults
+merge = False
+project = False
+compress = True
+delete = False
+
+outfolder = args.destination
+csv_file = args.csv
+if args.name:
+    final_name = args.name
+    merge = True
+if args.crs:
+    projection = args.crs
+    project = True
+if args.dont_compress:
+    compress = False
+
+# outfolder = 'c:\\gis\\elevation\\statewide\\test'
+# csv_file = 'c:\\users\\jdadams\\downloads\\10mDEMS2.csv'
+# projection = "EPSG:26912"
+# final_name = "10m"
+
+ext_list = ('.img', '.tif', '.asc', '.jpg', '.jgw', '.xyz', '.tfw', '.dbf', 
+            '.prj', '.shp', '.shx')
+raster_exts = ('.img', '.tif', '.asc', '.jpg')
+
+# merge = True
+# project = True 
+# compress = True  # Default to always compress rasters w/ lzw
+
+# TODO: Need to set these automagically. Maybe something in CSV?
+raster = True
+vector = False
+
+
+# Progress bar for download
+# modified from https://sumit-ghosh.com/articles/python-download-progress-bar/
+def progbar(progress, total):
+    done = int(50 * progress / total)
+    percent = round(100 * progress / total, 2)
+    sys.stdout.write('\r[{}{}] {}%'.format('#' * done, '_' * (50 - done),
+                                           percent))
+    sys.stdout.flush()
+
+
+# Shamelessly modified from Sumit Ghosh, 
+# https://sumit-ghosh.com/articles/python-download-progress-bar/
```

Comment:
is there a license for this that needs to be satisfied or is it also MIT or the like?

Reply:
As much as I can find, there is no explicit license. The code is published to a github gist w/o a license, and the blogpost says "Hope this comes in useful for you."

Reply:
no license means copyright unfortunately. I did notice that sentence which implies it might be CC? As long as it's not copy paste then I think it's ok.